examples/demo/src/products/ProductReferenceField.tsx

```diff
@@ -1,7 +1,8 @@
-import React from 'react';
+import React, { FC } from 'react';
 import { ReferenceField, TextField } from 'react-admin';
+import { FieldProps } from './../types';
 
-const ProductReferenceField = props => (
+const ProductReferenceField: FC<FieldProps & { resource: string }> = props => (
```

Comment:
`resource` should be added to the `FieldProps` type, it's always passed by react-admin.